### DISCUSSION TITLE
Create a model and repository without ID property in Loopback 4

### DISCUSSION THREAD

**Discussion by alvis-do** [loopbackio/loopback-next#7822] [Q&A]
Hi everyone, I have a problem with creating a model and repository without ID property in Loopback 4.

Most models have an ID attribute, so I can create a repository extends from the `DefaultCrudRepository` class. This class required an ID attribute. So, it works only for any models that have an ID attribute. 

![image](https://user-images.githubusercontent.com/38392756/131632596-eb0d509f-2980-4f84-a002-b931f7b6cc8e.png)

But, in some cases, I have a model without an ID attribute and I need to create a repository for this model. In this case, I have no idea about what the class of repository that I need to extend is.

Does Anybody have experience with loopback 4 to help me? Thank all.

**Comment by achrinza** [comment]
Have you tried swapping that ID attribute with `undefined`?
